doc/read-the-docs-site/extensions-flags-pragmas.rst

```diff
@@ -0,0 +1,83 @@
+At this time, it is not possible to make function applications non-strict: ``(\(~x) -> 42) (3 + 4)`` still evaluates ``3 + 4`` strictly.
+
+Making let bindings strict by default has the following advantages:
+
+* It makes let bindings and function applications semantically equivalent, e.g., ``let x = 3 + 4 in 42`` has the same semantics as ``(\x -> 42) (3 + 4)``.
+  This is what one would come to expect, as it is the case in most other programming languages, regardless of whether the language is strict or non-strict.
+* Untyped Plutus Core programs, which are compiled from Plutus Tx, are not evaluated lazily (unlike Haskell), that is, there is no memoization of the results of evaluated expressions.
+  Thus using non-strict bindings can cause an expression to be inadvertently evaluated for an unbounded number of times.
+  Consider ``let x = <expensive> in \y -> x + y``.
+  If ``x`` is non-strict, ``<expensive>`` will be evalutated every time ``\y -> x + y`` is applied to an argument, which means it can be evaluated 0 time, 1 time, 2 times, or any number of times (this is not the case if lazy evaluation was employed).
+  On the other hand, if ``x`` is strict, it is always evaluated once, which is at most one more time than what is necessary.
+
+Flags
+-------------------------------------------------
+
+GHC has a variety of optimization flags, many of which are on by default.
+Although Plutus Tx is, syntactically, a subset of Haskell, it has different semantics and a different evaluation strategy (Haskell: non-strict semantics, call by need; Plutus Tx: strict semantics, call by value).
+As a result, some GHC optimizations are not helpful for Plutus Tx programs, and can even be harmful, in the sense that it can make Plutus Tx programs less efficient, or fail to be compiled.
+An example is the full laziness optimization, controlled by GHC flag ``-ffull-laziness``, which floats let bindings out of lambdas whenever possible.
+Since Untyped Plutus Core does not employ lazy evaluation, the full laziness optimization is never beneficial, and can sometimes make a Plutus Tx program more expensive.
+Conversely, some GHC features must be turned on in order to ensure Plutus Tx programs are compiled successfully.
+
+All Plutus Tx modules should use the following GHC flags: ::
+
+  -fplugin PlutusTx.Plugin
+  -fno-ignore-interface-pragmas
+  -fno-omit-interface-pragmas
+  -fno-full-laziness
+  -fno-spec-constr
+  -fno-specialise
+  -fno-strictness
+  -fno-unbox-strict-fields
+  -fno-unbox-small-strict-fields
+
+``-fplugin PlutusTx.Plugin`` enables the Plutus Tx compiler.
+``-fno-ignore-interface-pragmas`` and ``-fno-omit-interface-pragmas`` ensure unfoldings of Plutus Tx functions are available.
+The rest are GHC optimizations that are bad for Plutus Tx, and should thus be turned off.
+
+Note that this section only covers GHC flags, not Plutus Tx compiler flags.
+Information about the latter can be found in :ref:`plutus_tx_options`.
```

Comment:
Clicking on the link from the rendered file doesn't lead me anywhere.

Reply:
This does not render properly on GitHub (although it's not too bad). To see the properly rendered text you'd need to check out the branch, run `autobuild-docs` and look at the HTML.

Reply:
Meh, do folks actually do that? I read all the docs on GitHub.

Reply:
Most things do render nicely, except a few things like links.